controllers/data.controller.js

```diff
@@ -378,6 +378,8 @@ exports.getUserRanking = async (req, res) => {
   try {
     // get user from db with user ID then if user found check if he is in the page, if not, include him
     const users = await User.find();
+    const currentUsername = req.query.username;
+    const user = users.find((user) => user.username === currentUsername);
```

Comment:
I have not checked it before. Would it be better to query some length of users like 50~100 users per api call. And slice with 10 to make pagination in frontend page?